test/discord.test.ts

```diff
@@ -293,161 +294,407 @@ describe('calling getUserId', () => {
     });
 
-    it('should send error message and log when command is "!tasks due" and error ' 
-        + 'getting project ID', done => {
-        expect.assertions(2);
-            
-        const client = setupClient();
-
-        const projectId = 0;
-        const error = 'error';
-        const channelName = 'channel';
-
-        let sentMessageValue = '';
-
-        const loggerMock = new LoggerMockBuilder()
-            .withWarn(jest.fn().mockImplementation(value => {
-                expect(value).toBe(`Error getting tasks due for week: Error: ${error}`);
-                expect(sentMessageValue).toBe(`Unable to find ActiveCollab project for channel ` 
-                    + channelName);
-                done();
-            }))
-            .build();
-
-        const mappingControllerMock = new MappingControllerMockBuilder()
-            .withGetProjectId(jest.fn(() => { throw Error(error); }))
-            .build();
-
-        const discordController = new DiscordControllerBuilder()
-            .withMappingController(mappingControllerMock)
-            .withClient(client)
-            .withLogger(loggerMock)
-            .build();
-
-        const sentMessage = {
-            edit: jest.fn(async value => {
-                sentMessageValue = value;
-            })
-        };
-
-        const message = {
-            content: '!tasks due',
-            author: 'author',
-            channel: {
-                send: jest.fn(async () => sentMessage),
-                startTyping: jest.fn(() => Promise.resolve()),
-                stopTyping: jest.fn(() => Promise.resolve()),
-                type: 'text',
-                name: channelName
-            }
-        };
-
-        client.emit('message', message);
+    describe('when message is "!tasks in <list>"', () => {
+        it('should call commandController.taskInListForProject with list from '
+        +   'command and project ID from channel', done => {
+            const client = setupClient();
+
+            const taskList = 'Selected for Development';
+            const projectId = 0;
+            const returnedTasks = new RichEmbed();
+
+            const commandControllerMock = new CommandControllerMockBuilder()
+                .withTasksInListForProject(jest.fn(() => Promise.resolve(returnedTasks)))
+                .build();
+
+            const mappingControllerMock = new MappingControllerMockBuilder()
+                .withGetProjectId(jest.fn().mockReturnValue(projectId))
+                .build();
+
+            const discordController = new DiscordControllerBuilder()
+                .withClient(client)
+                .withMappingController(mappingControllerMock)
+                .withCommandController(commandControllerMock)
+                .build();
+
+            const message = new MessageBuilder()
+                .withContent(`!tasks in ${taskList}`)
+                .withSend(jest.fn(async value => {
+                    if (value !== returnedTasks) {
```

Comment:
The test should probably fail here if the callback was called with something unexpected.